src/main/java/org/apache/maven/buildcache/BuildCacheMojosExecutionStrategy.java

```diff
@@ -434,6 +440,55 @@ private static String normalizedPath(Path path, Path baseDirPath) {
         return normalizedPath;
     }
 
+    /**
+     * Filters array values based on ignore pattern and converts to string representation.
+     */
+    private static String filterAndStringifyArray(Object array, String ignorePattern) {
+        if (ignorePattern == null) {
+            return ArrayUtils.toString(array);
+        }
+
+        java.util.regex.Pattern pattern = java.util.regex.Pattern.compile(ignorePattern);
```

Comment:
ignore patterns will be of a small and finite count. please consider to cache them.